presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java

```diff
@@ -75,15 +75,13 @@
 import static java.util.Objects.requireNonNull;
 import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
 import static java.util.concurrent.TimeUnit.MILLISECONDS;
-import static java.util.concurrent.TimeUnit.SECONDS;
 import static javax.ws.rs.core.MediaType.TEXT_PLAIN_TYPE;
 import static javax.ws.rs.core.Response.Status.NOT_FOUND;
 
 @Path("/v1/statement/executing")
 public class ExecutingStatementResource
 {
     private static final Logger log = Logger.get(ExecutingStatementResource.class);
-    private static final Duration MAX_WAIT_TIME = new Duration(1, SECONDS);
```

Comment:
@rzeyde-varada if we could change the default, what value would you use (and why)?

Reply:
I would prefer to allow the client to set it per-query, since long-running queries can be polled less frequently.
For example, if a query takes O(1 minute), it should probably be fine to limit long polling `maxWait` to 10 seconds - reducing the polling overhead by ~90%.

Reply:
I think, the client can set this already

https://github.com/prestosql/presto/blob/6a53550134f4dfd4563a3b9a8d9afa8b6b3c9411/presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java#L154

> it should probably be fine to set long polling maxWait to 10 seconds - reducing the polling overhead by ~90%.

@electrum would there be a problem if we set MAX_WAIT_TIME to 10s by default? 
(to be 1s seems pretty low number)

Reply:
That changes the stats update interval for CLI and other clients.

Reply:
We can keep the default `maxWait` at 1s, but limit the client-provided `maxWait` only if it's above 10s. WDYT?

Reply:
I’m not sure I understand the proposal.

Reply:
Sorry, I'll try to re-phrase:
If the client doesn't provide `maxWait`, we'll use a "default" value from the configuration (e.g. `query.max-wait.default`).
If the client provides `maxWait` value and it's above the "limit" (e.g. `query.max-wait.max`), we'll use the limit instead of the client-provided `maxWait`.
https://github.com/prestosql/presto/pull/3844/files#diff-783c05a894231fbb9b92d1689442935fL206-R210
```diff
-        Duration wait = WAIT_ORDERING.min(MAX_WAIT_TIME, maxWait);
+        Duration wait = executingStatementConfig.getMaxWaitDefault();
+        if (maxWait != null) {
+            wait = WAIT_ORDERING.min(executingStatementConfig.getMaxWaitLimit(), maxWait);
+        }
```